portable/ThirdParty/GCC/RP2040/library.cmake

```diff
@@ -0,0 +1,67 @@
+# Called after the Raspberry Pi Pico SDK has been initialized to add our libraries
+
+add_library(FreeRTOS-Kernel_included INTERFACE)
+target_compile_definitions(FreeRTOS-Kernel_included INTERFACE
+        -DFREERTOS_KERNEL=1
+        # todo don't know if we need this; maybe just want to explicitly disable things
+        -DPICO_RTOS=1 # let SDK know that RTOS is in use
+        )
+
+pico_add_platform_library(FreeRTOS-Kernel_included)
+
+add_library(FreeRTOS-Kernel-Core INTERFACE)
+target_sources(FreeRTOS-Kernel-Core INTERFACE
+        ${FREERTOS_KERNEL_PATH}/croutine.c
+        ${FREERTOS_KERNEL_PATH}/event_groups.c
+        ${FREERTOS_KERNEL_PATH}/list.c
+        ${FREERTOS_KERNEL_PATH}/queue.c
+        ${FREERTOS_KERNEL_PATH}/stream_buffer.c
+        ${FREERTOS_KERNEL_PATH}/tasks.c
+        ${FREERTOS_KERNEL_PATH}/timers.c
+        )
+target_include_directories(FreeRTOS-Kernel-Core INTERFACE ${FREERTOS_KERNEL_PATH}/include)
+
+add_library(FreeRTOS-Kernel INTERFACE)
+target_sources(FreeRTOS-Kernel INTERFACE
+        ${CMAKE_CURRENT_LIST_DIR}/port.c
```

Comment:
This should probably be named something port specific.

Reply:
yeah i mean the namespace is what the user build has included. i.e. they chose to include the RP2040 port of FreeRTOS. We would need a more specific name if installed, but I think that could be handled by namespacing during the packaging step (we don't use system installed libraries yet with the SDK)